nilearn/tests/test_surface.py

```diff
@@ -0,0 +1,393 @@
+        true_locations = [vertex + offsets for vertex in mesh[0]]
+        assert_array_almost_equal(true_locations, locations)
+    assert_raises(ValueError, surface._sample_locations,
+                  mesh, affine, 1., kind='bad_kind')
+
+
+def test_masked_indices():
+    mask = np.ones((4, 3, 8))
+    mask[:, :, ::2] = 0
+    locations = np.mgrid[:5, :3, :8].ravel().reshape((3, -1))
+    masked = surface._masked_indices(locations.T, mask.shape, mask)
+    # These elements are masked by the mask
+    assert_true((masked[::2] == 1).all())
+    # The last element of locations is one row beyond first image dimension
+    assert_true((masked[-24:] == 1).all())
+    # 4 * 3 * 8 / 2 elements should remain unmasked
+    assert_true((1 - masked).sum() == 48)
+
+
+def test_projection_matrix():
+    mesh = _flat_mesh(5, 7, 4)
+    img = _z_const_img(5, 7, 13)
+    proj = surface.projection_matrix(
+        mesh, np.eye(4), img.shape, radius=2., n_points=10)
+    # proj matrix has shape (n_vertices, img_size)
+    assert_equal(proj.shape, (5 * 7, 5 * 7 * 13))
+    # proj.dot(img) should give the values of img at the vertices' locations
+    values = proj.dot(img.ravel()).reshape((5, 7))
+    assert_array_almost_equal(values, img[:, :, 0])
+    mesh = _flat_mesh(5, 7)
+    proj = surface.projection_matrix(
+        mesh, np.eye(4), (5, 7, 1), radius=.1, n_points=10)
+    assert_array_almost_equal(proj.toarray(), np.eye(proj.shape[0]))
+    mask = np.ones(img.shape, dtype=int)
+    mask[0] = 0
+    proj = surface.projection_matrix(
+        mesh, np.eye(4), img.shape, radius=2., n_points=10, mask=mask)
+    proj = proj.toarray()
+    # first row of the mesh is masked
+    assert_array_almost_equal(proj.sum(axis=1)[:7], np.zeros(7))
+    assert_array_almost_equal(proj.sum(axis=1)[7:], np.ones(proj.shape[0] - 7))
+    # mask and img should have the same shape
+    assert_raises(ValueError, surface.projection_matrix,
+                  mesh, np.eye(4), img.shape, mask=np.ones((3, 3, 2)))
+
+
+def test_sampling_affine():
+    # check sampled (projected) values on a toy image
+    img = np.ones((4, 4, 4))
+    img[1, :, :] = 2
+    nodes = [[1, 1, 2], [10, 10, 20], [30, 30, 30]]
+    mesh = [np.asarray(nodes), None]
+    affine = 10 * np.eye(4)
+    affine[-1, -1] = 1
+    texture = surface._nearest_voxel_sampling(
+        [img], mesh, affine=affine, radius=1, kind='ball')
+    assert_array_equal(texture[0], [1., 2., 1.])
+    if LooseVersion(scipy.__version__) < LooseVersion('0.14'):
+        return
+    texture = surface._interpolation_sampling(
+        [img], mesh, affine=affine, radius=0, kind='ball')
+    assert_array_almost_equal(texture[0], [1.1, 2., 1.])
+
+
+def test_sampling():
+    mesh = _flat_mesh(5, 7, 4)
+    img = _z_const_img(5, 7, 13)
+    mask = np.ones(img.shape, dtype=int)
+    mask[0] = 0
+    projectors = [surface._nearest_voxel_sampling]
+    if LooseVersion(scipy.__version__) >= LooseVersion('0.14'):
```

Comment:
Can you please comment for this version check ? Why we need version check ? Just to be clear for everybody